tests/features/test_covered_distance.py

```diff
@@ -0,0 +1,194 @@
+        self.assertDictEqual(output, self.expected_output_total_and_vel_distance)
+
+    # test covered distance, velocity interval, and acceleration interval
+    def test_get_total_and_vel_and_acc_distance(self):
+        input_df = self.input.copy()
+        vel_intervals = self.vel_intervals
+        acc_intervals = self.acc_intervals
+        input_df.drop(columns=["home_1_ax"], inplace=True)
+        with self.assertRaises(ValueError):
+            get_covered_distance(input_df, ["home_1","away_2"], self.framerate, velocity_intervals=vel_intervals, acceleration_intervals=acc_intervals)
+
+        input_df = self.input.copy()
+        output = get_covered_distance(input_df, ["home_1","away_2"], self.framerate, velocity_intervals=vel_intervals, acceleration_intervals=acc_intervals)
+        self.assertDictEqual(output, self.expected_output_total_and_vel_and_acc_distance)
+
+    # check input covered distance
+    def test_get_covered_distance_wrong_input(self):
+        
+        # tracking_data
+        with self.assertRaises(TypeError) as cm:
+            data = {"ball_x": [1, 2, 3, 4]}
+            get_covered_distance(data, ["home_1", "away_2"], 1)
+        self.assertEqual(
+            str(cm.exception), 
+            f"tracking data must be a pandas DataFrame, not a {type(data).__name__}"
+        )
+        
+        # player_ids
+        with self.assertRaises(TypeError) as cm:
+            players = "home_1"
+            get_covered_distance(self.input, players, 1)
+        self.assertEqual(
+            str(cm.exception), 
+            f"player_ids must be a list, not a {type(players).__name__}"
+        )
+
+        with self.assertRaises(TypeError) as cm:
+            players = ["home_1", 123]
+            get_covered_distance(self.input, [players], 1)
+        self.assertEqual(
+            str(cm.exception), 
+            "All elements in player_ids must be strings"
+        )
+        
+        # framerate
+        with self.assertRaises(TypeError) as cm:
+            framerate = "1"
+            get_covered_distance(self.input, ["home_1", "away_2"], framerate)
+        self.assertEqual(
+            str(cm.exception), 
+            f"framerate must be a int, not a {type(framerate).__name__}"
+        )
+
+    # test check intervals function
+    def test_parse_intervals(self):
+        # check type
+        intervals = (0, 15.0, 90, 'a', 3, -1)
+        with self.assertRaises(TypeError) as cm:
+            _parse_intervals(intervals)
+        self.assertEqual(
+            str(cm.exception), 
+            "All elements in the tuple must be integers or floats"
+        )
+
+        # check even number
+        intervals = (0, 15, 90, 3, -1)
+        with self.assertRaises(ValueError) as cm:
+            _parse_intervals(intervals)
+        self.assertEqual(
+            str(cm.exception), 
+            "Intervals must contain an even number of elements."
+        )
```

Comment:
We might have to rethink how we want to include the intervals. I think either of 2 options would be more logical

1. tuples in tuples: `((0, 2), (2, 4), (4, 6), (6, np.inf))`

2. continous intervals: `(0, 2, 4, 6, np.inf) # results in the same as above`

Think this is more logical than putting everying in one list and creating intervals from that. Maybe it would be even nice to have support for both options (if a tuple/list contains only floats, it is second option, if only tuples/lists the first, otherwise raise an error.)